cmake/CPM.cmake

```diff
@@ -0,0 +1,210 @@
+# TheLartians/CPM - A simple Git dependency manager
+# =================================================
+# See https://github.com/TheLartians/CPM for usage and update instructions.
+#
+# MIT License
+# -----------
+#[[
+  Copyright (c) 2019 Lars Melchior
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy
+  of this software and associated documentation files (the "Software"), to deal
+  in the Software without restriction, including without limitation the rights
+  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+
+  The above copyright notice and this permission notice shall be included in all
+  copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+  SOFTWARE.
+]]
+
+cmake_minimum_required(VERSION 3.10 FATAL_ERROR)
+
+set(CURRENT_CPM_VERSION 0.11.1)
+
+if(CPM_DIRECTORY)
+  if(NOT ${CPM_DIRECTORY} MATCHES ${CMAKE_CURRENT_LIST_DIR})
+    if (${CPM_VERSION} VERSION_LESS ${CURRENT_CPM_VERSION})
+      CPM_HANDLE_OLD_VERSION(${CURRENT_CPM_VERSION})
+    endif()
+    return()
+  endif()
+endif()
+
+set(CPM_VERSION ${CURRENT_CPM_VERSION} CACHE INTERNAL "")
+set(CPM_DIRECTORY ${CMAKE_CURRENT_LIST_DIR} CACHE INTERNAL "")
+set(CPM_PACKAGES "" CACHE INTERNAL "")
+
+option(CPM_USE_LOCAL_PACKAGES "Use locally installed packages (find_package)" OFF)
```

Comment:
Would it be better to default this to ON? I kind of like the idea of using local dependencies by default.